sklearn/linear_model/_ridge.py

```diff
@@ -1064,8 +1064,21 @@ def predict(self, y_predict):
         return y_predict
 
 
+class _IdentityClassifier(LinearClassifierMixin):
+    """Fake classifier which will directly output the prediction.
+
+    We inherit from LinearClassifierMixin to get the proper shape for the
+    output `y`.
+    """
+    def __init__(self, classes):
+        self.classes_ = classes
+
+    def decision_function(self, y_predict):
+        return y_predict
```

Comment:
This should probably be:

```python
def decision_function(self, X):
    assert X.shape[1] == 1
    return X[:, 0]
```

Reply:
Why we should limit to first class only?

The `y_predict` will be from the prediction computed in Ridge: 

```python
predictions = y - (c / G_inverse_diag)
```

so it will be as many columns as classes. And it seems that `decision_function` in `LinearClassifierMixin` will do something similar regarding the shape of `y`, isn't, it?